<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from bigman.srdf.xacro              | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!--This does not replace URDF, and is not an extension of URDF.
    This is a format for representing semantic information about the robot structure.
    A URDF file must exist for this robot as well, where the joints and the links that are referenced are defined
-->
<robot name="bigman" xmlns:xacro="http://ros.org/wiki/xacro">
  <!--GROUPS: Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
  <!--LINKS: When a link is specified, the parent joint of that link (if it exists) is automatically included-->
  <!--JOINTS: When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
  <!--CHAINS: When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
  <!--SUBGROUPS: Groups can also be formed by referencing to already defined group names-->
  <group name="base">
    <link name="Waist"/>
  </group>
  <group name="force_torque_sensors">
    <joint name="l_leg_ft_joint"/>
    <joint name="r_leg_ft_joint"/>
    <joint name="l_arm_ft_joint"/>
    <joint name="r_arm_ft_joint"/>
  </group>
  <group name="left_arm">
    <joint name="l_arm_ft_joint"/>
    <chain base_link="DWYTorso" tip_link="l_wrist"/>
  </group>
  <group name="right_arm">
    <joint name="r_arm_ft_joint"/>
    <chain base_link="DWYTorso" tip_link="r_wrist"/>
  </group>
  <group name="left_leg">
    <joint name="pippo"/> <!-- Just for test -->
    <joint name="l_leg_ft_joint"/>
    <chain base_link="Waist" tip_link="l_sole"/>
  </group>
  <group name="right_leg">
    <joint name="r_leg_ft_joint"/>
    <chain base_link="Waist" tip_link="r_sole"/>
  </group>
  <group name="torso">
    <chain base_link="Waist" tip_link="torso"/>
  </group>
  <group name="chains">
    <!--THIS HAS TO BE THE LAST GROUP IN THE FILE -->
    <group name="left_arm"/>
    <group name="right_arm"/>
    <group name="left_leg"/>
    <group name="right_leg"/>
    <group name="torso"/>
  </group>
</robot>

